examples/misc/packed_bubbles.py

```diff
@@ -0,0 +1,163 @@
+            rest_bub = np.delete(self.bubbles, i, 0)
+            # try to move directly towoards the center of mass
+            # dir_vec from bubble to com
+            dir_vec = self.com - self.bubbles[i, :2]
+
+            # shorten dir_vec to have length of 1
+            dir_vec = dir_vec / np.sqrt(dir_vec.dot(dir_vec))
+
+            # calculate new bubble position
+            new_point = self.bubbles[i, :2] + dir_vec * self.step_dist
+            new_bubble = np.append(new_point, self.bubbles[i, 2:4])
+
+            # check whether new bubble collides with oder bubbles
+            if not self.check_collisions(new_bubble, rest_bub):
+                self.bubbles[i, :] = new_bubble
+                self.com = self.center_of_mass()
+                moves += 1
+            else:
+                # try to move around a bubble that you collide with
+                # find colliding bubble
+                for colliding in self.collides_with(new_bubble, rest_bub):
+                    # calculate dir vec
+                    dir_vec = rest_bub[colliding, :2] - self.bubbles[i, :2]
+                    dir_vec = dir_vec / np.sqrt(dir_vec.dot(dir_vec))
+                    # calculate orthagonal vec
+                    orth = np.array([dir_vec[1], -dir_vec[0]])
+                    # test which dir to go
+                    new_point1 = self.bubbles[i, :2] + orth * self.step_dist
+                    new_point2 = self.bubbles[i, :2] - orth * self.step_dist
+                    dist1 = self.center_distance(
+                        self.com, np.array([new_point1]))
+                    dist2 = self.center_distance(
+                        self.com, np.array([new_point2]))
+                    new_point = new_point1 if dist1 < dist2 else new_point2
+                    new_bubble = np.append(new_point, self.bubbles[i, 2:4])
+                    if not self.check_collisions(new_bubble, rest_bub):
+                        self.bubbles[i, :] = new_bubble
+                        self.com = self.center_of_mass()
+
+        if moves / len(self) < 0.1:
+            self.step_dist = self.step_dist / 2
+
+    def plot(self, ax, labels, colors):
+        """
+        draw the bubble plot
+
+        Args:
+            ax (matplotlib.axes._subplots.AxesSubplot)
+            labels (list): labels of the bubbles
+            colors (list): colors of the bubbles
+        """
+        for i in range(len(self)):
+            circ = plt.Circle(
+                self.bubbles[i, :2], self.bubbles[i, 2], color=colors[i])
+            ax.add_patch(circ)
+            ax.text(*self.bubbles[i, :2], labels[i],
+                    horizontalalignment='center', verticalalignment='center')
+
+
+# set market share of the browsers as area of the bubbles
+bubble_plot = BubbleChart(a=np.array(
+    browser_market_share['market_share']), bubble_distance=1)
+
+# collapse plot 50 times. In some cases it might be useful
+# to do this more or less often
+for i in range(50):
```

Comment:
As an API design, could this call be `bubble_plot.collapse(n_iterations=50)`?  It seems nicer to internalize "this is an iterative process" in the process its self.  And while I don't think this is worth doing for the example, it would also let you encapsulate logic like "go N interactions or until it stops changing _much_" in the collapse method rather than on the user side.